app/collector.go

```diff
@@ -110,15 +110,11 @@ func (c *collector) Report(_ context.Context) (report.Report, error) {
 	}
 	c.reports = clean(c.reports, c.window)
 
-	rpt := report.MakeReport()
-	id := murmur3.New64()
+	rpts := []report.Report{}
 	for _, tr := range c.reports {
-		rpt = rpt.Merge(tr.report)
-		id.Write([]byte(tr.report.ID))
+		rpts = append(rpts, tr.report)
 	}
```

Comment:
If you split timestampReport into two arrays (one for timestamps, the other for reports, keeping them in sync), you could eliminate this loop, and copying all the reports. That might not be that expensive, though, as there should only be ~15 of them?